### ISSUE TITLE
Font optimization breaks dynamic font loading

### ISSUE THREAD

**Issue by kripod** [vercel/next.js#20134]
# Bug report

Font optimization has been enabled in the canary releases of Next.js, by #19758. Since then, Google Fonts stylesheets cannot be injected in runtime with `next/head`.

## Describe the bug

I'm using the gist below for loading Google Fonts on demand:

https://gist.github.com/kripod/52c0f37ad6c4d5e84b4e3a7bc976f50b

Usage example:

```tsx
<GoogleFontsSheet
  fonts={{
    "Source Sans Pro": [
      { wght: 400 },
      { wght: 400, ital: 1 },
      { wght: 700 },
      { wght: 700, ital: 1 },
    ],
    "Playfair Display": [{ wght: 700 }],
  }}
/>
```

It generates the following code:

```tsx
import Head from "next/head";

<Head>
  <link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet" />
</Head>
```

However, when passing user-changeable data to the `fonts` prop of `<GoogleFontsSheet>`, the corresponding style sheets refuse to load.

## To Reproduce

Please see the instructions above.

## Expected behavior

Runtime injection of Google Fonts via `<link>` elements should be allowed. The optimization shouldn't remove elements which load fonts dynamically.

## System information

- OS: macOS
- Version of Next.js: 10.0.4-canary.5
- Version of Node.js: 14.15.0
- Deployment: next start

**Comment by rishi-raj-jain** [comment]
Is this what you're looking forward to https://codesandbox.io/s/tender-pike-vyybw?file=/pages/index.js ?

**Comment by kripod** [comment]
Unfortunately, this is still an issue, and Next.js 10.2 has been released with this bug inside. That’s how dynamic font loading looks now (trying to inject `<link>` tags on the fly):

<img width="371" alt="Uniform-looking fonts" src="https://user-images.githubusercontent.com/14854048/116593188-1f5aee80-a921-11eb-833f-ca9988f53ea5.png">

**Comment by kripod** [comment]
This seems to have been resolved by #26608.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
